go/api/config_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+
+package api
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/glide-for-redis/go/glide/protobuf"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDefaultStandaloneConfig(t *testing.T) {
+	config := NewRedisClientConfiguration()
+	expected := &protobuf.ConnectionRequest{
+		TlsMode:            protobuf.TlsMode_NoTls,
+		ClusterModeEnabled: false,
+		ReadFrom:           protobuf.ReadFrom_Primary,
+	}
+
+	result := config.toProtobuf()
+
+	assert.Equal(t, expected, result)
+}
+
+func TestDefaultClusterConfig(t *testing.T) {
+	config := NewRedisClusterClientConfiguration()
+	expected := &protobuf.ConnectionRequest{
+		TlsMode:            protobuf.TlsMode_NoTls,
+		ClusterModeEnabled: true,
+		ReadFrom:           protobuf.ReadFrom_Primary,
+	}
+
+	result := config.toProtobuf()
+
+	assert.Equal(t, expected, result)
+}
+
+func TestConfig_allFieldsSet(t *testing.T) {
+	hosts := []string{"host1", "host2"}
```

Comment:
Is there a way to make parametrized tests in go?

Reply:
Yep, see [here](https://gist.github.com/zaky/8304b5abbdc03600c29e?permalink_comment_id=4321142#gistcomment-4321142). I used this pattern for TestNodeAddress and TestRedisCredentials

Reply:
Can you parametrize this test then too?

Reply:
I could, but it will probably be difficult to read, mainly because the creation of the config and expected connection request where all fields are set requires quite a few lines of code. Let me know if you'd like me to make the change anyways.

Reply:
No need then